packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Square, DotsThreeVertical, X } from '@phosphor-icons/react';
+import React, { type HTMLAttributes, type KeyboardEvent, useEffect, useRef, useState } from 'react';
+
+import { Button, DropdownMenu, Input } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { getNext, getParent, getPrevious, getItems, type Item, getLastDescendent } from './types';
+
+// TODO(burdon): Break/join lines.
+// TODO(burdon): TextObject/MarkdownEditor
+// TODO(burdon): Cut-and-Paste.
+// TODO(burdon): Drag-and-drop.
+// TODO(burdon): Better key nav.
+
+type TreeOptions = Pick<HTMLAttributes<HTMLInputElement>, 'placeholder' | 'spellCheck'> & {
+  checkbox?: boolean;
+};
+
+//
+// Item
+//
+
+type ItemProps = {
+  item: Item;
+  active?: boolean;
```

Comment:
`active` is generally reserved for DnD – could this be called something else?

Reply:
Oh I thought you were using active to represent the current position in the table where the cursor was.

Reply:
No, that’s focus, and because it’s focus in particular that has to be handled outside of React. Some related concepts handled within React include `selected` and `current`, are those relevant here?

Reply:
Here I do mean focus so will change accordingly.